pkg/operation/cloudbotanist/vagrantbotanist/infrastructure.go

```
@@ -0,0 +1,83 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vagrantbotanist
+
+import (
+	"fmt"
+
+	"path/filepath"
+
+	"github.com/gardener/gardener/pkg/client/vagrant"
+	"github.com/gardener/gardener/pkg/operation/common"
+	pb "github.com/gardener/gardener/pkg/vagrantprovider"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeployInfrastructure talks to the gardener-vagrant-provider which creates the nodes.
+func (b *VagrantBotanist) DeployInfrastructure() error {
+
+	// TODO: use b.Operation.ComputeDownloaderCloudConfig("vagrant")
+	// At this stage we don't have the shoot api server
+	chart, err := b.Operation.ChartSeedRenderer.Render(filepath.Join(common.ChartPath, "shoot-cloud-config", "charts", "downloader"), "shoot-cloud-config-downloader", metav1.NamespaceSystem, map[string]interface{}{
+		"kubeconfig": string(b.Operation.Secrets["cloud-config-downloader"].Data["kubeconfig"]),
+		"secretName": b.Operation.Shoot.ComputeCloudConfigSecretName("vagrant"),
+	})
+	if err != nil {
+		return err
+	}
+
+	var cloudConfig = ""
+	for fileName, chartFile := range chart.Files {
+		if fileName == "downloader/templates/cloud-config.yaml" {
+			cloudConfig = chartFile
+		}
+	}
+
+	client, conn, err := vagrant.New(fmt.Sprintf(b.Shoot.Info.Spec.Cloud.Vagrant.Endpoint))
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	_, err = client.Start(context.Background(), &pb.StartRequest{
+		Cloudconfig: cloudConfig,
+		Id:          1,
+	})
+
+	return err
+}
+
+// DestroyInfrastructure talks to the gardener-vagrant-provider which destroys the nodes.
+func (b *VagrantBotanist) DestroyInfrastructure() error {
+	client, conn, err := vagrant.New(fmt.Sprintf(b.Shoot.Info.Spec.Cloud.Vagrant.Endpoint))
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	_, err = client.Delete(context.Background(), &pb.DeleteRequest{
```

Any reason why this error is ignored?

---

fixed